### PR TITLE
Add ability to create custom redirects for documentation.

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR adds the ability to create custom redirects for documentation. The rules only apply for pages that URL path starts with `/docs/`.

The change covers the following redirect rules:
| Legacy URLs               |  Current URLs                   | 
|----------------------------------------------------------------------|---------------------------------------------------------------|
| /docs/hyperformula     |  [/hyperformula](https://handsontable.github.io/hyperformula) | 
| /docs/hyperformula/whatever/whatever     |  [/hyperformula/whatever/whatever](https://handsontable.github.io/hyperformula/whatever/whatever) | 
| /docs/tutorial-introduction.html         |  /docs/                            | 
| /docs/tutorial-compatibility.html        |  /docs/supported-browsers          | 
| /docs/tutorial-licensing.html            |  /docs/software-license            | 
| /docs/tutorial-license-key.html          |  /docs/license-key                 | 
| /docs/tutorial-quick-start.html          |  /docs/installation                | 
| /docs/tutorial-data-binding.html         |  /docs/binding-to-data             | 
| /docs/tutorial-data-sources.html         |  /docs/binding-to-data             | 
| /docs/tutorial-load-and-save.html        |  /docs/saving-data                 | 
| /docs/tutorial-setting-options.html      |  /docs/setting-options             | 
| /docs/tutorial-grid-sizing.html          |  /docs/grid-size                   | 
| /docs/tutorial-using-callbacks.html      |  /docs/events-and-hooks            | 
| /docs/tutorial-keyboard-navigation.html  |  /docs/keyboard-navigation         | 
| /docs/tutorial-internationalization.html |  /docs/internationalization-i18n   | 
| /docs/tutorial-modules.html              |  /docs/modules                     | 
| /docs/tutorial-custom-build.html         |  /docs/building                    | 
| /docs/tutorial-custom-plugin.html        |  /docs/plugins                     | 
| /docs/tutorial-cell-types.html           |  /docs/cell-type                   | 
| /docs/tutorial-cell-editor.html          |  /docs/cell-editor                 | 
| /docs/tutorial-cell-function.html        |  /docs/cell-function               | 
| /docs/tutorial-suspend-rendering.html    |  /docs/batch-operations            | 
| /docs/tutorial-testing.html              |  /docs/testing                     | 
| /docs/tutorial-performance-tips.html     |  /docs/performance                 | 
| /docs/tutorial-release-notes.html        |  /docs/release-notes               | 
| /docs/tutorial-migration-guide.html      |  /docs/migration-from-7.4-to-8.0   | 
| /docs/demo-scrolling.html                |  /docs/row-virtualization          | 
| /docs/demo-fixing.html                   |  /docs/column-freezing             | 
| /docs/demo-resizing.html                 |  /docs/column-width                | 
| /docs/demo-moving.html                   |  /docs/column-moving               | 
| /docs/demo-pre-populating.html           |  /docs/row-prepopulating           | 
| /docs/demo-stretching.html               |  /docs/column-width                | 
| /docs/demo-freezing.html                 |  /docs/column-freezing             | 
| /docs/demo-fixing-bottom.html            |  /docs/row-freezing                | 
| /docs/demo-hiding-rows.html              |  /docs/row-hiding                  | 
| /docs/demo-hiding-columns.html           |  /docs/column-hiding               | 
| /docs/demo-trimming-rows.html            |  /docs/row-trimming                | 
| /docs/demo-bind-rows-headers.html        |  /docs/row-header                  | 
| /docs/demo-collapsing-columns.html       |  /docs/column-groups               | 
| /docs/demo-nested-headers.html           |  /docs/column-groups               | 
| /docs/demo-nested-rows.html              |  /docs/row-parent-child            | 
| /docs/demo-dropdown-menu.html            |  /docs/column-menu                 | 
| /docs/demo-sorting.html                  |  /docs/row-sorting                 | 
| /docs/demo-multicolumn-sorting.html      |  /docs/row-sorting                 | 
| /docs/demo-searching.html                |  /docs/searching-values            | 
| /docs/demo-filtering.html                |  /docs/column-filter               | 
| /docs/demo-summary-calculations.html     |  /docs/column-summary              | 
| /docs/demo-data-validation.html          |  /docs/cell-validator              | 
| /docs/demo-auto-fill.html                |  /docs/autofill-values             | 
| /docs/demo-merged-cells.html             |  /docs/merge-cells                 | 
| /docs/demo-alignment.html                |  /docs/text-alignment              | 
| /docs/demo-read-only.html                |  /docs/disabled-cells              | 
| /docs/demo-disabled-editing.html         |  /docs/disabled-cells              | 
| /docs/demo-custom-renderers.html         |  /docs/cell-renderer               | 
| /docs/demo-numeric.html                  |  /docs/numeric-cell-type           | 
| /docs/demo-date.html                     |  /docs/date-cell-type              | 
| /docs/demo-time.html                     |  /docs/time-cell-type              | 
| /docs/demo-checkbox.html                 |  /docs/checkbox-cell-type          | 
| /docs/demo-select.html                   |  /docs/select-cell-type            | 
| /docs/demo-dropdown.html                 |  /docs/dropdown-cell-type          | 
| /docs/demo-autocomplete.html             |  /docs/autocomplete-cell-type      | 
| /docs/demo-password.html                 |  /docs/password-cell-type          | 
| /docs/demo-handsontable.html             |  /docs/handsontable-cell-type      | 
| /docs/demo-context-menu.html             |  /docs/context-menu                | 
| /docs/demo-spreadsheet-icons.html        |  /docs/icon-pack                   | 
| /docs/demo-comments_.html                |  /docs/comments                    | 
| /docs/demo-copy-paste.html               |  /docs/basic-clipboard             | 
| /docs/demo-export-file.html              |  /docs/export-to-csv               | 
| /docs/demo-conditional-formatting.html   |  /docs/conditional-formatting      | 
| /docs/demo-customizing-borders.html      |  /docs/formatting-cells            | 
| /docs/demo-selecting-ranges.html         |  /docs/selection                   | 
| /docs/demo-formula-support.html          |  /docs/formula-calculation         | 
| /docs/react                              |  /docs/react-installation          | 
| /docs/angular                            |  /docs/angular-installation        | 
| /docs/vue3                               |  /docs/vue3-installation           | 
| /docs/vue                                |  /docs/vue-installation            | 
| /docs/i18n/missing-language-code         |  /docs/internationalization        | 
| /docs/frameworks-wrapper-for-react-simple-examples.html               |  /docs/react-simple-examples               | 
| /docs/frameworks-wrapper-for-react-...                                |  /docs/react-...                           | 
| /docs/frameworks-wrapper-for-vue-installation.html                    |  /docs/vue-installation                    | 
| /docs/frameworks-wrapper-for-vue-...                                  |  /docs/vue-...                             | 
| /docs/frameworks-wrapper-for-angular-installation.html                |  /docs/angular-installation                | 
| /docs/frameworks-wrapper-for-angular-...                              |  /docs/angular-...                         | 
| /docs/9.0/tutorial-compatibility.html                                     |  /docs/9.0/supported-browsers                  | 
| /docs/9.0/tutorial-licensing.html                                         |  /docs/9.0/software-license                    | 
| /docs/9.0/tutorial-license-key.html                                       |  /docs/9.0/license-key                         | 
| /docs/9.0/tutorial-quick-start.html                                       |  /docs/9.0/installation                        | 
| /docs/9.0/tutorial-data-binding.html                                      |  /docs/9.0/binding-to-data                     | 
| /docs/9.0/tutorial-data-sources.html                                      |  /docs/9.0/binding-to-data                     | 
| /docs/9.0/tutorial-load-and-save.html                                     |  /docs/9.0/saving-data                         | 
| /docs/9.0/tutorial-setting-options.html                                   |  /docs/9.0/setting-options                     | 
| /docs/9.0/tutorial-grid-sizing.html                                       |  /docs/9.0/grid-size                           | 
| /docs/9.0/tutorial-using-callbacks.html                                   |  /docs/9.0/events-and-hooks                    | 
| /docs/9.0/tutorial-keyboard-navigation.html                               |  /docs/9.0/keyboard-navigation                 | 
| /docs/9.0/tutorial-internationalization.html                              |  /docs/9.0/internationalization-i18n           | 
| /docs/9.0/tutorial-modules.html                                           |  /docs/9.0/modules                             | 
| /docs/9.0/tutorial-custom-build.html                                      |  /docs/9.0/building                            | 
| /docs/9.0/tutorial-custom-plugin.html                                     |  /docs/9.0/plugins                             | 
| /docs/9.0/tutorial-cell-types.html                                        |  /docs/9.0/cell-type                           | 
| /docs/9.0/tutorial-cell-editor.html                                       |  /docs/9.0/cell-editor                         | 
| /docs/9.0/tutorial-cell-function.html                                     |  /docs/9.0/cell-function                       | 
| /docs/9.0/tutorial-suspend-rendering.html                                 |  /docs/9.0/batch-operations                    | 
| /docs/9.0/tutorial-testing.html                                           |  /docs/9.0/testing                             | 
| /docs/9.0/tutorial-performance-tips.html                                  |  /docs/9.0/performance                         | 
| /docs/9.0/tutorial-release-notes.html                                     |  /docs/9.0/release-notes                       | 
| /docs/9.0/tutorial-migration-guide.html                                   |  /docs/9.0/migration-from-7.4-to-8.0           | 
| /docs/9.0/demo-scrolling.html                                             |  /docs/9.0/row-virtualization                  | 
| /docs/9.0/demo-fixing.html                                                |  /docs/9.0/column-freezing                     | 
| /docs/9.0/demo-resizing.html                                              |  /docs/9.0/column-width                        | 
| /docs/9.0/demo-moving.html                                                |  /docs/9.0/column-moving                       | 
| /docs/9.0/demo-pre-populating.html                                        |  /docs/9.0/row-prepopulating                   | 
| /docs/9.0/demo-stretching.html                                            |  /docs/9.0/column-width                        | 
| /docs/9.0/demo-freezing.html                                              |  /docs/9.0/column-freezing                     | 
| /docs/9.0/demo-fixing-bottom.html                                         |  /docs/9.0/row-freezing                        | 
| /docs/9.0/demo-hiding-rows.html                                           |  /docs/9.0/row-hiding                          | 
| /docs/9.0/demo-hiding-columns.html                                        |  /docs/9.0/column-hiding                       | 
| /docs/9.0/demo-trimming-rows.html                                         |  /docs/9.0/row-trimming                        | 
| /docs/9.0/demo-bind-rows-headers.html                                     |  /docs/9.0/row-header                          | 
| /docs/9.0/demo-collapsing-columns.html                                    |  /docs/9.0/column-groups                       | 
| /docs/9.0/demo-nested-headers.html                                        |  /docs/9.0/column-groups                       | 
| /docs/9.0/demo-nested-rows.html                                           |  /docs/9.0/row-parent-child                    | 
| /docs/9.0/demo-dropdown-menu.html                                         |  /docs/9.0/column-menu                         | 
| /docs/9.0/demo-sorting.html                                               |  /docs/9.0/row-sorting                         | 
| /docs/9.0/demo-multicolumn-sorting.html                                   |  /docs/9.0/row-sorting                         | 
| /docs/9.0/demo-searching.html                                             |  /docs/9.0/searching-values                    | 
| /docs/9.0/demo-filtering.html                                             |  /docs/9.0/column-filter                       | 
| /docs/9.0/demo-summary-calculations.html                                  |  /docs/9.0/column-summary                      | 
| /docs/9.0/demo-data-validation.html                                       |  /docs/9.0/cell-validator                      | 
| /docs/9.0/demo-auto-fill.html                                             |  /docs/9.0/autofill-values                     | 
| /docs/9.0/demo-merged-cells.html                                          |  /docs/9.0/merge-cells                         | 
| /docs/9.0/demo-alignment.html                                             |  /docs/9.0/text-alignment                      | 
| /docs/9.0/demo-read-only.html                                             |  /docs/9.0/disabled-cells                      | 
| /docs/9.0/demo-disabled-editing.html                                      |  /docs/9.0/disabled-cells                      | 
| /docs/9.0/demo-custom-renderers.html                                      |  /docs/9.0/cell-renderer                       | 
| /docs/9.0/demo-numeric.html                                               |  /docs/9.0/numeric-cell-type                   | 
| /docs/9.0/demo-date.html                                                  |  /docs/9.0/date-cell-type                      | 
| /docs/9.0/demo-time.html                                                  |  /docs/9.0/time-cell-type                      | 
| /docs/9.0/demo-checkbox.html                                              |  /docs/9.0/checkbox-cell-type                  | 
| /docs/9.0/demo-select.html                                                |  /docs/9.0/select-cell-type                    | 
| /docs/9.0/demo-dropdown.html                                              |  /docs/9.0/dropdown-cell-type                  | 
| /docs/9.0/demo-autocomplete.html                                          |  /docs/9.0/autocomplete-cell-type              | 
| /docs/9.0/demo-password.html                                              |  /docs/9.0/password-cell-type                  | 
| /docs/9.0/demo-handsontable.html                                          |  /docs/9.0/handsontable-cell-type              | 
| /docs/9.0/demo-context-menu.html                                          |  /docs/9.0/context-menu                        | 
| /docs/9.0/demo-spreadsheet-icons.html                                     |  /docs/9.0/icon-pack                           | 
| /docs/9.0/demo-comments_.html                                             |  /docs/9.0/comments                            | 
| /docs/9.0/demo-copy-paste.html                                            |  /docs/9.0/basic-clipboard                     | 
| /docs/9.0/demo-export-file.html                                           |  /docs/9.0/export-to-csv                       | 
| /docs/9.0/demo-conditional-formatting.html                                |  /docs/9.0/conditional-formatting              | 
| /docs/9.0/demo-customizing-borders.html                                   |  /docs/9.0/formatting-cells                    | 
| /docs/9.0/demo-selecting-ranges.html                                      |  /docs/9.0/selection                           | 
| /docs/9.0/demo-formula-support.html                                       |  /docs/9.0/formula-calculation                 | 
| /docs/9.0/frameworks-wrapper-for-react-installation.html                  |  /docs/9.0/react-installation                  | 
| /docs/9.0/frameworks-wrapper-for-react-...                                |  /docs/9.0/react-...                           | 
| /docs/9.0/frameworks-wrapper-for-vue-installation.html                    |  /docs/9.0/vue-installation                    | 
| /docs/9.0/frameworks-wrapper-for-vue-...                                  |  /docs/9.0/vue-...                             | 
| /docs/9.0/frameworks-wrapper-for-angular-installation.html                |  /docs/9.0/angular-installation                | 
| /docs/9.0/frameworks-wrapper-for-angular-...                              |  /docs/9.0/angular-...                         | 

_[skip changelog]_

### How has this been tested?

The changes can be tested at https://dev.handsontable.com/docs.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Affected project(s):
- [x] `docs`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: docs/docker/redirects.conf [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# Some examples:
+#  - Redirect the page permanently (using HTTP 301 code) from
+#    https://handsontable.com/docs/api/very-old-collection to https://handsontable.com/docs/api/new-collection
+# (uncomment the line bellow to use the rule)
+#rewrite ^/api/very-old-collection/?$ /docs/api/new-collection/ permanent;
+#
+#  - Redirect the page temporary (using HTTP 302 code) from
+#    https://handsontable.com/docs/11.0/api/temp-collection to https://handsontable.com/docs/11.0/api/other-page
+# (uncomment the line bellow to use the rule)
+#rewrite ^/11.0/api/temp-collection/?$ /docs/11.0/api/other-page/ permanent;
```
Comment (warpech) [inline docs/docker/redirects.conf:18]: I guess that we already have some manual redirection rules. Can they be moved to here for clarity?

File: docs/docker/redirects.conf [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+# Some examples:
+#  * Redirect the page permanently (using HTTP 301 code) from
+#    https://handsontable.com/docs/api/very-old-collection to https://handsontable.com/docs/api/new-collection
+# (uncomment the line bellow to use the rule)
+#rewrite ^/api/very-old-collection/?$ /docs/api/new-collection/ permanent;
+#
+#  * Redirect the page temporary (using HTTP 302 code) from
+#    https://handsontable.com/docs/11.0/api/temp-collection to https://handsontable.com/docs/11.0/api/other-page
+# (uncomment the line bellow to use the rule)
+#rewrite ^/11.0/api/temp-collection/?$ /docs/11.0/api/other-page/ permanent;
```
Comment (kirszenbaum) [inline docs/docker/redirects.conf:18]:
```suggestion
# This file lists Nginx rewrite rules for the documentation pages:
# (i.e. pages whose URL paths start with /docs/).
# 
# Usage examples:
#  - To redirect permanently (using the HTTP 301 code)
#    from https://handsontable.com/docs/api/very-old-collection
#    to https://handsontable.com/docs/api/new-collection
#    add the following, uncommented line:
#    rewrite ^/api/very-old-collection/?$ /docs/api/new-collection/ permanent;
#
#  - Redirect temporarily (using the HTTP 302 code)
#    from https://handsontable.com/docs/11.0/api/temp-collection
#    to https://handsontable.com/docs/11.0/api/other-page
#    add the following, uncommented line:
#    rewrite ^/11.0/api/temp-collection/?$ /docs/11.0/api/other-page/ permanent;
#
# Read more:
# - Creating Nginx rewrite rules: https://www.nginx.com/blog/creating-nginx-rewrite-rules/
# - The Ngix rewrite module docs: http://nginx.org/en/docs/http/ngx_http_rewrite_module.html
```

File: docs/docker/redirects.conf [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+# --- HyperFormula ---
+rewrite ^/hyperformula$ https://handsontable.github.io/hyperformula permanent;
+rewrite ^/hyperformula/(.*)$ https://handsontable.github.io/hyperformula/$1 permanent;
+
+# --- HoT logs that in the console ---
+rewrite ^/i18n/missing-language-code /docs/internationalization-i18n/#loading-the-prepared-language-files permanent;
+
+# --- frameworks shortcut ---
```
Comment (kirszenbaum) [inline docs/docker/redirects.conf]:
```suggestion
# --- framework shortcuts ---
```

File: docs/docker/redirects.conf [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+#  * Redirect the page temporary (using HTTP 302 code) from
+#    https://handsontable.com/docs/11.0/api/temp-collection to https://handsontable.com/docs/11.0/api/other-page
+# (uncomment the line bellow to use the rule)
+#rewrite ^/11.0/api/temp-collection/?$ /docs/11.0/api/other-page/ permanent;
+
+# --- HyperFormula ---
+rewrite ^/hyperformula$ https://handsontable.github.io/hyperformula permanent;
+rewrite ^/hyperformula/(.*)$ https://handsontable.github.io/hyperformula/$1 permanent;
+
+# --- HoT logs that in the console ---
```
Comment (kirszenbaum) [inline docs/docker/redirects.conf:24]:
```suggestion
# --- documentation links in Handsontable's console logs ---
```

**Review by kirszenbaum — CHANGES_REQUESTED** [review]
@budnix - this looks great, thanks a lot!

**Review by kirszenbaum — APPROVED** [review]

**Review by kirszenbaum — APPROVED** [review]

**Comment by budnix** [comment]
@warpech 🏓

**Comment by warpech** [comment]
@kirszenbaum have you tested it in the following way?

- broken links checker reports 404 on some page
- you fix that 404 by redirecting the dead link to some other page in `redirects.conf` (I am not saying this will always be the solution, but sometimes yes)
- broken links checker no longer reports the 404

**Comment by kirszenbaum** [comment]
> @kirszenbaum have you tested it in the following way?
> 
> * broken links checker reports 404 on some page
> * you fix that 404 by redirecting the dead link to some other page in `redirects.conf` (I am not saying this will always be the solution, but sometimes yes)
> * broken links checker no longer reports the 404

@warpech, @budnix No, I didn't test it, just corrected the content added.

I now tried this:
1. `npm run docs:build`.
2. The broken links report shows, for example, this link:
    
3. I add the following rule:
    `rewrite ^/10.0/vue3-installation/$ /docs/10.0/vue-installation/ permanent;`
4. `npm run docs:build`
5. The link is still coming up in the broken links report:
     

Am I doing sth wrong?

**Comment by budnix** [comment]
> 
> 1. `npm run docs:build`.
> 2. The broken links report shows, for example, this link:
>    
> 3. I add the following rule:
>    `rewrite ^/10.0/vue3-installation/$ /docs/10.0/vue-installation/ permanent;`
> 4. `npm run docs:build`
> 5. The link is still coming up in the broken links report:
>    
> 
> Am I doing sth wrong?

These kinds of redirects work on staging or on production only. You can try to create a separate branch with changes and deploy it manually using [Docs Staging Deployment](https://github.com/handsontable/handsontable/actions/workflows/docs-staging.yml) workflow. Within the "Run workflow" button select a newly created branch and click "Run". Another tip is that Chrome likes to cache the rewrites so if you're sure that the rule is correct but it still doesn't work try incognito mode.

**Comment by kirszenbaum** [comment]
> > 1. `npm run docs:build`.
> > 2. The broken links report shows, for example, this link:
> >    
> > 3. I add the following rule:
> >    `rewrite ^/10.0/vue3-installation/$ /docs/10.0/vue-installation/ permanent;`
> > 4. `npm run docs:build`
> > 5. The link is still coming up in the broken links report:
> >    
> > 
> > Am I doing sth wrong?
> 
> These kinds of redirects work on staging or on production only. You can try to create a separate branch with changes and deploy it manually using [Docs Staging Deployment](https://github.com/handsontable/handsontable/actions/workflows/docs-staging.yml) workflow. Within the "Run workflow" button select a newly created branch and click "Run". Another tip is that Chrome likes to cache the rewrites so if you're sure that the rule is correct but it still doesn't work try incognito mode.

@budnix Thanks, makes sense!

I deployed the changes to Staging (from a [`feature/docs-manual-redirects-test` branch](https://github.com/handsontable/handsontable/blob/feature/docs-manual-redirects-test/docs/docker/redirects.conf#L67)), but it doesn't seem to work: https://dev.handsontable.com/docs/10.0/vue3-installation/ (should redirect to: https://dev.handsontable.com/docs/10.0/vue-installation/). Tried resetting cache and incognito

**Comment by warpech** [comment]
So if I understand correctly, the broken links checker spawns its own dumb HTTP server (https://github.com/handsontable/handsontable/blob/5c392c3ec12e1684d9735ef1e20541b46ac8ff5d/docs/.vuepress/tools/check-links.js#L10) instead of using our Docker image. If we managed to run the Docker image locally, the broken links checker could use that instead, including all of the redirections.

@budnix do you confirm?

Anyway, that appears to be out of scope here. @kirszenbaum if you agree that this is worth pursuing, please make a new issue about that and add a label "Tools".

**Review by warpech — APPROVED** [review]
Approving the idea, I did not test

**Comment by budnix** [comment]
> So if I understand correctly, the broken links checker spawns its own dumb HTTP server 
instead of using our Docker image. If we managed to run the Docker image locally, the broken links checker could use that instead, including all of the redirections.
> @budnix do you confirm?

True, The tool checks the local VuePress instance without checking the internal Nginx redirects. 

Once I've merged the PR I thought that it's maybe a good time o rethink how we define the redirects? The Nginx redirects are implemented to handle URLs from the previous documentation engine and mostly for handling current numbered URLs to non-numbered (`/11.0/foo` -> `/foo`). This approach allows us to redirect users in the most performant way, as fast as possible. However, there is a second alternative. We can define a list of rewrites in the VuePress conf file like:
```js
export default ({ router }) => {
  router.addRoutes([
    { path: '/foo/', redirect: '/' },
    { path: '/bar/', redirect: '/' }
  ])
}
```
it's more secure than messing with Nginx configuration (low level) but less performant.

**Comment by warpech** [comment]
If I understand correctly, routes added via `addRoutes` work on the client so (among other problems) they cannot change the HTTP status code. I think that is not a valid replacement for Nginx, then.

If the only problem that we have with Nginx is that we are afraid messing with its config files, I hope that it is largely mitigated by our broken links checker.

**Pull request merged**